flink-core/src/main/java/org/apache/flink/core/fs/SafetyNetCloseableRegistry.java

```diff
@@ -134,10 +135,17 @@ public void close() throws IOException {
 				--GLOBAL_SAFETY_NET_REGISTRY_COUNT;
 				if (0 == GLOBAL_SAFETY_NET_REGISTRY_COUNT) {
 					REAPER_THREAD.interrupt();
```

Comment:
I am wondering whether we shouldn't join on the reaper thread when the last `CloseableRegistry` closes? From a resource lifecycle management one could argue that the reaper thread belongs to the `CloseableRegistries` and when the last closes, then we must also make sure that the thread properly terminates.

Reply:
All reclamation related classes and methods, say, `CloseableReaperThread`, `PhantomDelegatingCloseableRef` and `SafetyNetCloseableRegistry(Supplier<CloseableReaperThread> reaperThreadSupplier)`, are private or visible for testing, normal clients should care only about the reclamation of leaking resources guarantee exported by `SafetyNetCloseableRegistry`, "join on the reaper thread" does not belong to that exported guarantee.

Reply:
Alright. It would also be out of the scope of this PR to change it.